### ISSUE TITLE
Improve clarity on encoding errors

### ISSUE THREAD

**Issue by krisbitney** [goncalossilva/kotlinx-resources#47]
I check if the resource exists, the check returns true, I try to read the file, and I get a null pointer exception.

readTestResource(/Users/kris/IdeaProjects/krisbitney/polywrap-kt/src/commonTest/kotlin/helpers.kt
```kotlin
fun readTestResource(filePath: String): Result<ByteArray> {
    val resourceRoot = "src/commonTest/resources";
    val resource = Resource("$resourceRoot/$filePath")
    return if (resource.exists()) {
        val bytes = resource.readText().encodeToByteArray() // LINE 7; CAUSES NULL POINTER EXCEPTION
        Result.success(bytes)
    } else {
        Result.failure(Exception("File not found at $filePath."));
    }
}
```

Stack trace:
```
kotlin.NullPointerException
	at kotlin.Throwable#<init>(/opt/buildAgent/work/460cf706e11bbdb0/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Throwable.kt:29)
	at kotlin.Exception#<init>(/opt/buildAgent/work/460cf706e11bbdb0/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:21)
	at kotlin.RuntimeException#<init>(/opt/buildAgent/work/460cf706e11bbdb0/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:32)
	at kotlin.NullPointerException#<init>(/opt/buildAgent/work/460cf706e11bbdb0/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:43)
	at <global>.ThrowNullPointerException(/opt/buildAgent/work/460cf706e11bbdb0/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/RuntimeUtils.kt:15)
	at com.goncalossilva.resources.Resource#readText(/Users/runner/work/kotlinx-resources/kotlinx-resources/resources-library/src/darwinMain/kotlin/Resource.kt:17)
	at <global>.#readTestResource(/Users/kris/IdeaProjects/krisbitney/polywrap-kt/src/commonTest/kotlin/helpers.kt:7)
```

**Comment by goncalossilva** [comment]
Just to confirm @krisbitney, the file under `$resourceRoot/$filePath` exists, correct?

**Comment by goncalossilva** [comment]
I suspect this may be an encoding error. Would you mind checking the encoding of the file you're trying to read?

For example:

```
file -bi src/commonTest/resources/<path> # on macOS, use capital `I`
```

**Comment by krisbitney** [comment]
> Just to confirm @krisbitney, the file under `$resourceRoot/$filePath` exists, correct?

It exists and is successfully read on JVM and JS (Node)

**Comment by krisbitney** [comment]
> I suspect this may be an encoding error. Would you mind checking the encoding of the file you're trying to read?
> 
> For example:
> 
> ```
> file -bi src/commonTest/resources/<path> # on macOS, use capital `I`
> ```

It is encoded in the MessagePack format. It is not utf-8. Maybe that's the issue!

Have you considered adding a `readBytes` method?

**Comment by goncalossilva** [comment]
An initial version of reading binary data is done and released on 0.3.0 via `readBytes`. Give it a try and let me know what you think. :)
